src/main/scala/com/campudus/tableaux/database/model/TableauxModel.scala

```diff
@@ -69,15 +69,13 @@ sealed trait StructureDelegateModel extends DatabaseQuery {
   }
 }
 
-class TableauxModel(override protected[this] val connection: DatabaseConnection) extends DatabaseQuery with StructureDelegateModel {
+class TableauxModel(
+  override protected[this] val connection: DatabaseConnection,
+  override protected[this] val structureModel: StructureModel
```

Comment:
Yep, and this also told me that I should restructure or refactor the model creation — I don't want to say it but... `Singleton`.